src/test/java/com/linkedin/kafka/clients/utils/tests/AbstractKafkaIntegrationTestHarness.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils.tests;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import org.apache.kafka.common.protocol.SecurityProtocol;
+import org.testng.Assert;
+
+
+public abstract class AbstractKafkaIntegrationTestHarness extends AbstractZookeeperTestHarness {
+  protected Map<String, EmbeddedBroker> brokers = null;
+  protected String bootstrapUrl;
+
+  @Override
+  public void setUp() {
+    super.setUp();
+    if (brokers != null) {
+      return;
+    }
+
+    brokers = new LinkedHashMap<>();
+    List<Map<Object, Object>> brokerConfigs = buildBrokerConfigs();
+    Assert.assertNotNull(brokerConfigs);
+    Assert.assertFalse(brokerConfigs.isEmpty());
+    for (Map<Object, Object> brokerConfig : brokerConfigs) {
+      EmbeddedBroker broker = new EmbeddedBroker(brokerConfig);
+      String id = broker.getId();
+      if (brokers.putIfAbsent(id, broker) != null) {
+        KafkaTestUtils.quietly(broker::close); //wont be picked up by teardown
+        throw new IllegalStateException("multiple brokers defined with id " + id);
+      }
+    }
+
+    StringJoiner joiner = new StringJoiner(",");
+    brokers.values().forEach(broker -> joiner.add(broker.getAddr(securityProtocol())));
+    bootstrapUrl = joiner.toString();
+  }
+
+  @Override
+  public void tearDown() {
+    try {
+      if (brokers != null) {
+        for (EmbeddedBroker broker : brokers.values()) {
+          KafkaTestUtils.quietly(broker::close);
+        }
+        brokers.clear();
+        brokers = null;
+      }
+    } finally {
+      super.tearDown();
+    }
+  }
+
+  protected EmbeddedBroker serverForId(int id) {
+    return broker(Integer.toString(id));
+  }
+
+  protected EmbeddedBroker broker(String id) {
+    EmbeddedBroker broker = brokers.get(id);
+    if (broker == null) {
+      throw new IllegalArgumentException("Invalid server id " + id);
+    }
+    return broker;
+  }
+
+  public String bootstrapServers() {
+    return bootstrapUrl;
+  }
+
+
+  protected List<Map<Object, Object>> buildBrokerConfigs() {
```

Comment:
It might be useful to add documentation about this return value so the user knows that returnValue.get(i) is a config for the ith broker.